plugins/common/socket/stream.go

```diff
@@ -172,27 +173,33 @@ func (l *streamListener) setupConnection(conn net.Conn) error {
 }
 
 func (l *streamListener) closeConnection(conn net.Conn) {
+	// Fallback to enforce blocked reads on connections to end immediately
+	//nolint:errcheck // Ignore errors as this is a fallback only
+	conn.SetReadDeadline(time.Now())
+
 	addr := conn.RemoteAddr().String()
 	if err := conn.Close(); err != nil && !errors.Is(err, net.ErrClosed) && !errors.Is(err, syscall.EPIPE) {
 		l.Log.Warnf("Cannot close connection to %q: %v", addr, err)
+	} else {
+		l.Lock()
+		l.connections--
+		l.Unlock()
 	}
-	delete(l.connections, conn)
 }
 
 func (l *streamListener) address() net.Addr {
 	return l.listener.Addr()
 }
 
 func (l *streamListener) close() error {
-	if err := l.listener.Close(); err != nil {
-		return err
+	if l.listener != nil {
+		l.listener.Close()
```

Comment:
This will fail the linter if you rebase on master since you aren't checking the error right?